bgpd/bgp_vty.c

```diff
@@ -11004,85 +11004,171 @@ static int bgp_show_route_leak_vty(struct vty *vty, const char *name,
 	char *ecom_str;
 	vpn_policy_direction_t dir;
 
-	if (name) {
-		bgp = bgp_lookup_by_name(name);
+	if (use_json) {
+		json_object *json = NULL;
+		json_object *json_import_vrfs = NULL;
+		json_object *json_export_vrfs = NULL;
+
+		json = json_object_new_object();
+		json_import_vrfs = json_object_new_array();
+		json_export_vrfs = json_object_new_array();
+
+		/* Provide context for the block */
+		json_object_string_add(json, "vrf", name ? name : "default");
+		json_object_string_add(json, "afiSafi",
+					afi_safi_print(afi, safi));
+
+		bgp = name ? bgp_lookup_by_name(name) : bgp_get_default();
+
 		if (!bgp) {
-			vty_out(vty, "%% No such BGP instance exist\n");
+			json_object_boolean_true_add(json,
+						     "bgpNoSuchInstance");
+			vty_out(vty, "%s\n",
+				json_object_to_json_string_ext(
+					json,
+					JSON_C_TO_STRING_PRETTY));
+			json_object_free(json);
```

Comment:
Sorry, missed one, the 2 JSON arrays are leaked here too :disappointed:

Reply:
Please explain how the arrays are leaked at this point.  Since the arrays were added to the "json" object, my understanding is that when the "json" object is freed, the arrays will be as well. Is this incorrect?

Reply:
never mind.  I understand your point and have resolved it by moving the array creation.  Thanks for the review